### PR TITLE
Add new numeric variable types to C API and schema

### PR DESCRIPTION
**Pull request by t-sommer**
- add new fmi3[U]Int{8,16,32,64} and fmi3Float32 variable types
- replace fmi3Integer with fmi3Int32
- replace fmi3Real with fmi3Float64
- rename variable get / set functions accordingly
- remove "Real" from fmi3SetRealInputDerivatives and fmi3GetRealOutputDerivatives
- update XSD schema

### CODE REVIEW

File: headers/fmi3FunctionTypes.h [hunk 1]
```diff
@@ -407,13 +471,13 @@ typedef fmi3Status fmi3GetStatusTYPE       (fmi3Component c,
                                             const fmi3StatusKind s,
                                             fmi3Status* value);
 
-typedef fmi3Status fmi3GetRealStatusTYPE   (fmi3Component c,
+typedef fmi3Status fmi3GetDoubleStatusTYPE (fmi3Component c,
```
Comment (pmai) [inline headers/fmi3FunctionTypes.h]: Should this not be fmi3GetFloat64StatusTYPE? (We'll remove it later anyways if issue #484 goes the way we plan, but still).
```suggestion
typedef fmi3Status fmi3GetFloat64StatusTYPE (fmi3Component c,
```

File: headers/fmi3Functions.h [hunk 2]
```diff
@@ -149,13 +165,13 @@ Functions for FMI3 for Model Exchange
...
-#define fmi3SetRealInputDerivatives      fmi3FullName(fmi3SetRealInputDerivatives)
-#define fmi3GetRealOutputDerivatives     fmi3FullName(fmi3GetRealOutputDerivatives)
+#define fmi3SetInputDerivatives          fmi3FullName(fmi3SetInputDerivatives)
+#define fmi3GetOutputDerivatives         fmi3FullName(fmi3GetOutputDerivatives)
 #define fmi3DoStep                       fmi3FullName(fmi3DoStep)
 #define fmi3CancelStep                   fmi3FullName(fmi3CancelStep)
 #define fmi3GetStatus                    fmi3FullName(fmi3GetStatus)
-#define fmi3GetRealStatus                fmi3FullName(fmi3GetRealStatus)
-#define fmi3GetIntegerStatus             fmi3FullName(fmi3GetIntegerStatus)
+#define fmi3GetDoubleStatus              fmi3FullName(fmi3GetDoubleStatus)
```
Comment (pmai) [inline headers/fmi3Functions.h]: Dito for fmi3GetFloat64Status

```suggestion
#define fmi3GetFloat64Status              fmi3FullName(fmi3GetFloat64Status)
```

File: headers/fmi3Functions.h [hunk 3]
```diff
@@ -237,16 +270,16 @@ Functions for FMI3 for Co-Simulation
...
+FMI3_Export fmi3GetOutputDerivativesTYPE fmi3GetOutputDerivatives;
 
 FMI3_Export fmi3DoStepTYPE     fmi3DoStep;
 FMI3_Export fmi3CancelStepTYPE fmi3CancelStep;
 
 /* Inquire slave status */
 FMI3_Export fmi3GetStatusTYPE        fmi3GetStatus;
-FMI3_Export fmi3GetRealStatusTYPE    fmi3GetRealStatus;
-FMI3_Export fmi3GetIntegerStatusTYPE fmi3GetIntegerStatus;
+FMI3_Export fmi3GetDoubleStatusTYPE  fmi3GetDoubleStatus;
```
Comment (pmai) [inline headers/fmi3Functions.h]: And again:
```suggestion
FMI3_Export fmi3GetFloat64StatusTYPE  fmi3GetFloat64Status;
```

File: headers/fmi3FunctionTypes.h (hunk 1 above)
Comment (t-sommer) [reply headers/fmi3FunctionTypes.h]: @pmai, should we address this here? The functions will be superseded anyway (as decided today in the Steering Committee meeting).

Comment (pmai) [reply headers/fmi3FunctionTypes.h]: I'd do so for consistency's sake, makes the other change orthogonal...

**Pull request merged**